filesystems/blob/src/main/scala/cromwell/filesystems/blob/BlobPathBuilder.scala

```diff
@@ -67,19 +67,34 @@ class BlobPathBuilder(container: BlobContainerName, endpoint: EndpointURL)(priva
   override def name: String = "Azure Blob Storage"
 }
 
+object BlobPath {
+  // The Azure NIO library uses `{containerName}:` as the root of the path.
+  // This doesn't work well for our need to easily transfer back and forth
+  // to and from the blob URL format. This method removes anything up to and including
+  // the first colon, to create a path string useful for working with BlobPath.
+  // This is safe because the NIO library enforces no colons except to mark
+  // the root container name.
+  private def nioPathString(nioPath: NioPath): String = {
+    val pathStr = nioPath.toString
+    pathStr.substring(pathStr.indexOf(":")+1)
+  }
+}
+
 case class BlobPath private[blob](pathString: String, endpoint: EndpointURL, container: BlobContainerName)(private val fsm: BlobFileSystemManager) extends Path {
   override def nioPath: NioPath = findNioPath(pathString)
 
-  override protected def newPath(nioPath: NioPath): Path = BlobPath(nioPath.toString, endpoint, container)(fsm)
+  override protected def newPath(nioPath: NioPath): Path =
+    BlobPath(BlobPath.nioPathString(nioPath), endpoint, container)(fsm)
```

Comment:
Would it make sense to add:
```
  def apply(nioPath: NioPath, endpoint: EndpointURL, container: BlobContainerName)(fsm: BlobFileSystemManager): BlobPath = {
    BlobPath(nioPathString(nioPath), endpoint, container)(fsm)
  }
```
to the `BlobPath` companion object and then just do:
```suggestion
    BlobPath(nioPath, endpoint, container)(fsm)
```
?
Or just make the `BlobPath` case class take an `NioPath` property and provide a `pathString` property/function?

Reply:
Yes, that would be much better!